core/src/main/scala/kafka/network/RequestChannel.scala

```diff
@@ -260,17 +229,23 @@ class RequestChannel(val numProcessors: Int, val queueSize: Int) extends KafkaMe
 
   /** Send a response back to the socket server to be sent over the network */
   def sendResponse(response: RequestChannel.Response) {
+    if (isTraceEnabled) {
+      val requestHeader = response.request.header
+      val apiKey = ApiKeys.forId(requestHeader.apiKey)
```

Comment:
True. We could add another field to the context I guess, if you think it's worth it.